emcc.py

```diff
@@ -1603,6 +1603,11 @@ def save_intermediate(name=None, suffix='js'):
         link_opts = [] if debug_level >= 4 or shared.Settings.CYBERDWARF else ['-strip-debug'] # remove LLVM debug if we are not asked for it
         if not shared.Settings.ASSERTIONS:
           link_opts += ['-disable-verify']
+        else:
+          # when verifying, LLVM debug info has some tricky linking aspects, and llvm-link will
+          # disable the type map in that case. we added linking to opt, so we need to do
+          # something similar, which we can do with a param to opt
+          link_opts += ['-disable-debug-info-type-map']
```

Comment:
Ah, ok. that's surprising and a little bit scary, but I guess llvm-link is dumb enough that it's maybe not such a big deal.
A more interesting question then is, when we use upstream opt, without modifications, why don't more things break? Is that behavior behind a flag?